app/controllers/site_controller.rb

```diff
@@ -205,7 +210,10 @@ def search_all_genes
       delim = params[:search][:genes].include?(',') ? ',' : ' '
       raw_genes = params[:search][:genes].split(delim)
       @genes = sanitize_search_values(raw_genes).split(',').map(&:strip)
-
+      # limit gene search for performance reasons
+      if @genes.size > Gene::MAX_GENE_SEARCH
+        @genes = @genes.take(Gene::MAX_GENE_SEARCH)
+      end
```

Comment:
I'll bet there's a good reason, but I'm not following why here we don't set a message while in search_genes we do--is that becasue search_genes forwards to here?

Reply:
don't hold anything up on my account -- I assume this comment is  just a result of seeing rails paths that I have yet to fully trace myself

Reply:
it's done at the UI level for the global gene search.  There's a way you can short-circuit it for the autocomplete in a single study, but not the global one.  So the alert always shows on the global gene search.